app/test_main.py

```diff
@@ -1 +1,61 @@
-# write your code here
+import pytest
+from unittest import mock
+from typing import Callable
+from app.main import can_access_google_page
+
+
+class TestCanAccessGooglePage:
+    tested_url = "https://www.google.com"
+
+    @staticmethod
+    @pytest.fixture
+    def mocked_has_internet_connection() -> None:
+        with (mock.patch("app.main.has_internet_connection")
+              as mocked_internet_connection):
+            yield mocked_internet_connection
+
+    @staticmethod
+    @pytest.fixture
+    def mocked_valid_google_url() -> None:
+        with mock.patch(
+                "app.main.valid_google_url") as mocked_google_url:
+            yield mocked_google_url
+
+    def test_has_internet_connection_called(
+            self,
+            mocked_has_internet_connection: Callable
+    ) -> None:
+        can_access_google_page(self.tested_url)
+        mocked_has_internet_connection.assert_called_once()
+
+    def test_valid_google_url_called_with(
+            self,
+            mocked_valid_google_url: Callable
+    ) -> None:
+        can_access_google_page(self.tested_url)
+        mocked_valid_google_url.assert_called_once_with(self.tested_url)
+
```

Comment:
Using the `@mock.patch` decorator directly can simplify the fixture creation process and make the code more straightforward.